wasm2c/README.md

```diff
@@ -194,12 +216,15 @@ typedef void (*wasm_rt_funcref_t)(void);
 ```
 
 Next are the definitions for a table element. `func_type` is a function index
-as returned by `wasm_rt_register_func_type` described below.
+as returned by `wasm_rt_register_func_type` described below. `module_instance`
+is the pointer to the module instance taht should be passed in when the func is
+called.
 
 ```c
 typedef struct {
   uint32_t func_type;
   wasm_rt_funcref_t func;
+  void* module_instance;
```

Comment:
I wonder if `wasm_rt_funcref_t` should be defined to include all these of these elements... and then `wasm_rt_elem_t` could evolve to be tagged union of `wasm_rt_funcref_t` with other possible table inhabitants?

Reply:
That is basically where we went in the later reference-types PR... happy to bring that in earlier if you want it now. (I don't know if later proposals will need the tagged union -- for reference-types, we didn't find we needed it because the code always knows statically whether it's dealing with a funcref table or an externref table.)

Here's where we were headed in the reference-types wasm-rt.h -- how do you feel about this / do you want this here or wait until later?

```cpp
/**
 * A generic function pointer type, both for Wasm functions (`code`)
 * and host functions (`hostcode`). All function pointers are stored
 * in this canonical form, but must be cast to their proper signature
 * to call.
 */
typedef void (*wasm_rt_function_ptr_t)(void);

/** A function instance (the runtime representation of a function).
 * These can be stored in tables of type funcref, or used as values. */
typedef struct {
  /** The index as returned from `wasm_rt_register_func_type`. */
  uint32_t func_type;
  /** The function. The embedder must know the actual C signature of the
   * function and cast to it before calling. */
  wasm_rt_function_ptr_t func;
  /** A function instance is a closure of the function over an instance
   * of the originating module. The module_instance element will be passed into
   * the function at runtime. */
  void* module_instance;
} wasm_rt_funcref_t;

/** An external reference. These are opaque have meaning only to the host. */
typedef void* wasm_rt_externref_t;

/** Default (null) value of an externref */
static const wasm_rt_externref_t wasm_rt_externref_null_value = NULL;
```

Reply:
I think we can leave this for now